File: aoc2022/day1part1.py
import pathlib

inputfile = pathlib.Path('input-1.txt')
elves = []  # list of each elf's calories
elf_totcals = []  # list of each elf's total calories
with inputfile.open() as f:
    elves.append([]) 
    num_elves = len(elves) - 1
    for elf_cals in f:
        # reads each line including newline
        try:
            elf_cals = int(elf_cals.strip())
        except ValueError:
            # int('') raises exception, delimits elves
            elf_totcals.append(sum(elves[num_elves]))
            elves.append([])
            num_elves = len(elves) - 1
        else:
            elves[num_elves].append(elf_cals)
max_cals = max(elf_totcals)
elf_idx = elf_totcals.index(max_cals)
print(f'max cals: {max_cals} by elf #{elf_idx}')

File: aoc2022/day1part1.R
f <- file("input-1.txt")
l <- readLines(f)
close(f)

elf_cals <- 0  # current elf's calorie totoal
elves_cals <- list()  # list of elves calorie totals

for (idx in l) {
    if (idx != "") {
        elf_cals <- elf_cals + strtoi(idx)
    } else {
        elves_cals <- append(elves_cals, elf_cals)
        elf_cals <- 0
    }
}

max_cals <- max(unlist(elves_cals))
print(max_cals)